mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/InVmEventDeadLetters.java

```diff
@@ -0,0 +1,59 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import org.apache.james.mailbox.Event;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class InVmEventDeadLetters implements EventDeadLetters {
+
+    private final Multimap<Group, Event> deadLetters;
+
+    InVmEventDeadLetters() {
+        this.deadLetters = ArrayListMultimap.create();
```

Comment:
`/!\\` concurrency gonna mess that up.

As an exercise write simple concurrent tests as part of your contract.

Like 10 threads doing 100 insertions in 10 random group should be enough to trigger bugs.

A Synchronized multimap can be the object you look to.

Reply:
come later